Fix kickstart file error with user groups

When fill the "Group Membership" in "ADVANCED USER CONFIGURATION"
dialog with the provided example text:

wheel, my-team (1245), project-x (29935)

It keeps the spaces between group name(project-x) and group id('(29935)'),
and then write them to kickstart file. When boot anaconda with the kickstart
file, it fails with:

| unrecognzed arguments: (1245),project-x (29935)

Filter out the spaces between group name and group id to avoid such
issue.

Upstream-Status: Backport [https://github.com/rhinstaller/anaconda/commit/ae9cffd]

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
 pyanaconda/ui/gui/spokes/user.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyanaconda/ui/gui/spokes/user.py b/pyanaconda/ui/gui/spokes/user.py
index 3c8d2a7bb5..782e5721fd 100644
--- a/pyanaconda/ui/gui/spokes/user.py
+++ b/pyanaconda/ui/gui/spokes/user.py
@@ -156,7 +156,7 @@ class AdvancedUserDialog(GUIObject, GUIDialogInputCheckHandler):
             self.user.gid = None
 
         # ''.split(',') returns [''] instead of [], which is not what we want
-        self.user.groups = [g.strip() for g in self._tGroups.get_text().split(",") if g]
+        self.user.groups = [''.join(g.split()) for g in self._tGroups.get_text().split(",") if g]
 
         # Send ready signal to main event loop
         hubQ.send_ready(self.__class__.__name__, False)
-- 
2.25.1

